code/modules/mob/living/silicon/robot/inventory.dm

```diff
@@ -51,7 +52,9 @@
 		O.screen_loc = inv1.screen_loc
 		contents += O
 		if(istype(module_state_1,/obj/item/borg/sight))
-			sight_mode |= module_state_1:sight_mode
+			var/obj/item/borg/sight/S = module_state_1
+			sight_mode |= S.sight_mode
+			update_sight()

```

Comment:
calling update_sight so our vision is immediately updated when we equip the meson goggles module for example.